### PR TITLE
Tweak: Add `elementor-widget-wrap` to DOM optimization experiment (ED-2259)

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: assets/dev/js/editor/elements/views/column.js [hunk 1]
```diff
@@ -177,11 +183,13 @@ ColumnView = BaseElementView.extend( {
 		const emptyClass = 'elementor-element-empty',
 			populatedClass = 'elementor-element-populated';
 
-		if ( this.ui.columnInner ) {
+		const columnInner = ( this.ui.columnInner.length ) ? this.ui.columnInner : this.$el;
```
Comment (iNewLegend) [inline assets/dev/js/editor/elements/views/column.js:186]:
```
		const emptyClass = 'elementor-element-empty',
			populatedClass = 'elementor-element-populated';

		const columnInner = ( this.ui.columnInner.length ) ? this.ui.columnInner : this.$el;
```
can be
```
		const emptyClass = 'elementor-element-empty',
			populatedClass = 'elementor-element-populated',
			columnInner = ( this.ui.columnInner.length ) ? this.ui.columnInner : this.$el;
```

**Review by iNewLegend — CHANGES_REQUESTED** [review]

File: includes/elements/column.php [hunk 2]
```diff
@@ -200,21 +203,19 @@ protected function register_controls() {
...
-
 		$this->add_responsive_control(
 			'space_between_widgets',
 			[
 				'label' => __( 'Widgets Space', 'elementor' ) . ' (px)',
 				'type' => Controls_Manager::NUMBER,
 				'placeholder' => 20,
 				'selectors' => [
-					'{{WRAPPER}} ' . $space_between_widgets_selector . '> .elementor-widget-wrap > .elementor-widget:not(.elementor-widget__width-auto):not(.elementor-widget__width-initial):not(:last-child):not(.elementor-absolute)' => 'margin-bottom: {{VALUE}}px', //Need the full path for exclude the inner section
+					'{{WRAPPER}}' => '--widgets-gap: {{VALUE}}px',
```
Comment (matipojo) [inline includes/elements/column.php:218]: What will happen with `absolute` etc.

File: includes/elements/column.php [hunk 3]
```diff
@@ -939,11 +939,7 @@ public function before_render() {
 		$is_dom_optimization_active = Plugin::$instance->experiments->is_feature_active( 'e_dom_optimization' );
 		$wrapper_attribute_string = $is_dom_optimization_active ? '_widget_wrapper' : '_inner_wrapper';
 
-		$column_wrap_classes = $is_dom_optimization_active ? [ 'elementor-widget-wrap' ] : [ 'elementor-column-wrap' ];
-
-		if ( $this->get_children() ) {
-			$column_wrap_classes[] = 'elementor-element-populated';
```
Comment (matipojo) [inline includes/elements/column.php:945]: Why?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/elements/column.php (hunk 2 above)
Comment (StyleShit) [reply includes/elements/column.php:218]: Done.

File: assets/dev/js/editor/elements/views/column.js (hunk 1 above)
Comment (StyleShit) [reply assets/dev/js/editor/elements/views/column.js:186]: Done.

File: includes/elements/column.php (hunk 3 above)
Comment (StyleShit) [reply includes/elements/column.php:945]: Probably by mistake, nice catch :)

File: assets/dev/js/editor/elements/views/column.js [hunk 4]
```diff
@@ -33,6 +33,10 @@ ColumnView = BaseElementView.extend( {
 		var classes = BaseElementView.prototype.className.apply( this, arguments ),
 			type = this.isInner() ? 'inner' : 'top';
 
+		if ( elementorCommon.config.experimentalFeatures[ 'e_dom_optimization' ] ) {
```
Comment (matipojo) [inline assets/dev/js/editor/elements/views/column.js:36]:
```suggestion
		if ( ! elementorCommon.config.experimentalFeatures[ 'e_dom_optimization' ] ) {
```

**Review by matipojo — CHANGES_REQUESTED** [review]
Please see the comment + fix conflicts.

File: assets/dev/js/editor/elements/views/column.js (hunk 4 above)
Comment (StyleShit) [reply assets/dev/js/editor/elements/views/column.js:36]: Explain?

File: assets/dev/js/editor/document/hooks/ui/create/column-is-populated.js [hunk 5]
```diff
@@ -24,6 +24,7 @@ export class ColumnIsPopulated extends After {
 		containers.forEach( ( /* Container */ container ) => {
 			if ( 'column' === container.model.get( 'elType' ) ) {
 				container.view.changeChildContainerClasses();
+				container.view.moveResizeHandle();
```
Comment (iNewLegend) [inline assets/dev/js/editor/document/hooks/ui/create/column-is-populated.js:27]: Create a dedicated hook for __moveResizeHandle__, move logic from method __moveResizeHandle__ to the hook.

**Review by iNewLegend — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+			isDomOptimizationActive = elementorCommon.config.experimentalFeatures[ 'e_dom_optimization' ];
+
+		if ( ! isDomOptimizationActive ) {
+			return false;
+		}
+
+		// If the created element, was created at column.
+		return containers.some( ( /**Container*/ container ) => (
+			'column' === container.model.get( 'elType' )
+		) );
```
Comment (iNewLegend) [inline assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js:23]: Try replace with `getContainerType`.

**Comment by iNewLegend** [comment]
Great job

File: assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js (hunk 6 above)
Comment (StyleShit) [reply assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js]: Good idea! Forgot about that one 😅

**Review by iNewLegend — APPROVED** [review]

File: assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+	getConditions() {
+		// Trigger the hook only if DOM optimization experiment is active.
+		return elementorCommon.config.experimentalFeatures[ 'e_dom_optimization' ];
+	}
+
+	apply( args ) {
+		const { containers = [ args.container ] } = args;
+
+		containers.forEach( ( /* Container */ container ) => {
+			if ( 'column' === container.model.get( 'elType' ) ) {
```
Comment (matipojo) [inline assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js:25]: You already filtered to columns only by `getContainerType`.

File: assets/dev/js/editor/elements/views/column.js [hunk 8]
```diff
@@ -33,6 +33,12 @@ ColumnView = BaseElementView.extend( {
 		var classes = BaseElementView.prototype.className.apply( this, arguments ),
 			type = this.isInner() ? 'inner' : 'top';
 
+		// Add an `elementor-widget-wrap`class to the column in order to attach its styles & events to the column,
```
Comment (matipojo) [inline assets/dev/js/editor/elements/views/column.js:36]:
```suggestion
		// Add an `elementor-widget-wrap` class to the column in order to attach its styles & events to the column,
```

File: assets/dev/scss/frontend/_grid.scss [hunk 9]
```diff
@@ -191,25 +195,26 @@
 // Columns Gap
 .elementor-column {
 
-	> .elementor-element-populated {
+	&.elementor-element-populated,
+	.elementor-element-populated {
 
-		.elementor-column-gap-narrow  > & {
+		.elementor-column-gap-narrow & {
```
Comment (matipojo) [inline assets/dev/scss/frontend/_grid.scss:201]: What if the experiment is disabled?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js (hunk 7 above)
Comment (StyleShit) [reply assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js:25]: Yup, right :)

File: assets/dev/js/editor/elements/views/column.js (hunk 8 above)
Comment (StyleShit) [reply assets/dev/js/editor/elements/views/column.js:36]: Done.

File: assets/dev/scss/frontend/_grid.scss (hunk 9 above)
Comment (StyleShit) [reply assets/dev/scss/frontend/_grid.scss:201]: It works

File: assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
+import After from 'elementor-api/modules/hooks/ui/after';
+
+export class MoveColumnResizeHandle extends After {
```
Comment (kobizz) [inline assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js:3]: It shouldn't be applied by a hook. It should happen where you create the resize functionality

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/scss/frontend/_grid.scss [hunk 11]
```diff
@@ -191,25 +195,26 @@
...
 			padding: 5px;
 		}
 
-		.elementor-column-gap-default > & {
+		.elementor-column-gap-default & {
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply assets/dev/scss/frontend/_grid.scss:205]: Works properly :)

File: assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js (hunk 10 above)
Comment (matipojo) [reply assets/dev/js/editor/document/hooks/ui/create/move-column-resize-handle.js:3]: It's OK because it's a change from the normal flow.

**Review by matipojo — APPROVED** [review]

**Pull request merged**